ddtrace/internal/telemetry/metrics.py

```diff
@@ -37,43 +36,32 @@ def __init__(self, namespace, name, tags, common, interval=None):
         common: set to True if a metric is common to all tracers, false if it is python specific
         interval: field set for gauge and rate metrics, any field set is ignored for count metrics (in secs)
         """
-        self.name = name
+        self.name = name.lower()
         self.is_common_to_all_tracers = common
         self.interval = interval
-        self.namespace = namespace
-        self._tags = {}  # type: MetricTagType
-        self.set_tags(tags)
+        self.namespace = namespace.lower()
+        self._tags = {k.lower(): str(v).lower() for k, v in tags.items()}
```

Comment:
since we don't allow mutation of tags, we should consider just storing them as the final list of strings.

```python
self._tags = ["%s:%s" % (k, v) for k, v in tags.items()] if tags else []
```

this is a future improvement and not needed at this time.

Reply:
Great idea. Idk why I missed this 😓

Reply:
especially if any of the tags are static, they can be passed in as `key:value` and save the computation to join them